src/partr.c

```diff
@@ -481,16 +484,18 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *getsticky)
                 }
             }
             // the other threads will just wait for on signal to resume
-            int8_t gc_state = jl_gc_safe_enter(ptls);
             uv_mutex_lock(&ptls->sleep_lock);
             while (jl_atomic_load(&sleep_check_state) == sleeping) {
                 task = get_next_task(getsticky);
```

Comment:
This call also can't happen while holding the sleep lock (because it contains gc allocations and/or safepoints) and/or we cannot attempt to wake up all the other threads to get them to help with the gc effort.

Reply:
OK, in this case assuming get_next_task can call julia code/trigger GC then the `uv_mutex_lock(&ptls->sleep_lock);` must be done in GC safe region. Alternatively, all holders of the lock must not trigger any GC.

Reply:
It makes sense to me that we ought to be able to check for available tasks without triggering GC. Of course, that's hard to guarantee from julia code.